### PR TITLE
Move grpc common examples

### PR DESCRIPTION
**Pull request by stanley-cheung**
Move the Go `grpc-common` examples to the `grpc-go/examples` directory, preserving commit history.

End result is just to add 5 files to the `examples` directory

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+======================
+
+BACKGROUND
+-------------
+For this sample, we've already generated the server and client stubs from [helloworld.proto](https://github.com/grpc/grpc-common/blob/master/protos/helloworld.proto). 
+
+PREREQUISITES
+-------------
+
+- This requires Go 1.4
```
Comment (iamqizhao) [inline examples/README.md]: Go 1.4 or above?

Comment (stanley-cheung) [inline examples/README.md]: I was just copying the files from the `grpc-common` repo. Didn't make any change.

Comment (iamqizhao) [inline examples/README.md]: ok, this is not accurate now because 1.5 has been released. I can fix it later.

**Comment by iamqizhao** [comment]
Like the existing route_guide example, this greeter example should stay in a separate dir (e.g., examples/greeter or examples/helloworld).

File: examples/README.md (hunk 1 above)
Comment (dsymonds) [inline examples/README.md]: Uh, hopefully we're not going to immediately start requiring Go 1.5 for gRPC. It's polite to wait for a little while after a release before assuming everyone has upgraded.

Comment (iamqizhao) [inline examples/README.md]: @dsymonds I would change it to "Go 1.4 and above" instead of "requiring 1.5".

Comment (dsymonds) [inline examples/README.md]: Oh, yeah, that phrasing is fine.

**Comment by stanley-cheung** [comment]
Got it. Re-arranged the `helloworld` folder structure. Can you see if that makes sense now?

**Comment by iamqizhao** [comment]
Thanks!

**Pull request merged**